src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs

```diff
@@ -581,6 +581,38 @@ internal async Task<bool> CancelPreviousPoll(HttpContext context)
         }
     }
 
+    internal SetTransportState TrySetTransport(HttpTransportType transportType, HttpConnectionsMetrics metrics)
+    {
+        lock (_stateLock)
+        {
+            if (TransportType != HttpTransportType.None
+                && UseStatefulReconnect == false && TransportType != HttpTransportType.LongPolling)
+            {
+                return SetTransportState.AlreadyActive;
+            }
+
+            if (TransportType == HttpTransportType.None)
+            {
+                TransportType = transportType;
+
+                if (HttpConnectionsEventSource.Log.IsEnabled() || MetricsContext.ConnectionDurationEnabled)
+                {
+                    StartTimestamp = Stopwatch.GetTimestamp();
+                }
+
+                HttpConnectionsEventSource.Log.ConnectionStart(ConnectionId);
```

Comment:
If we're moving `Log.ConnectionStart` here, why not move `Log.ConnectionAlreadyActive` and `Log.CannotChangeTransport` here?

Reply:
This log is part of `HttpConnectionContext`. The other 2 are part of `HttpConnectionDispatcher`.